proposals/lip-0043.md

```diff
@@ -351,22 +351,23 @@ Transactions executing this command have:
 mainchainRegParams = {
     "type": "object",
     "required": [
-        "ownChainID",
         "ownName",
+        "ownChainID",
         "mainchainValidators",
+        "mainchainCertificateThreshold",
         "signature",
         "aggregationBits"
     ],
     "properties": {
-        "ownChainID": {
-            "dataType": "bytes",
-            "length": CHAIN_ID_LENGTH,
-            "fieldNumber": 1
-        },
         "ownName": {
             "dataType": "string",
             "minLength": MIN_CHAIN_NAME_LENGTH,
             "maxLength": MAX_CHAIN_NAME_LENGTH,
+            "fieldNumber": 1
+        },
+        "ownChainID": {
+            "dataType": "bytes",
+            "length": CHAIN_ID_LENGTH,
             "fieldNumber": 2
         },
         "mainchainValidators": {
```

Comment:
Can we add here:  `minItems: 1,
			maxItems: NUMBER_ACTIVE_VALIDATORS_MAINCHAIN,`

And remove: `   # mainchainValidators must have at least 1 element and at most NUMBER_ACTIVE_VALIDATORS_MAINCHAIN elements.
    if len(trs.params.mainchainValidators) < 1 or len(trs.params.mainchainValidators) > NUMBER_ACTIVE_VALIDATORS_MAINCHAIN:
raise Exception(f"Invalid mainchain validators array. It must have at least 1 element and at most {NUMBER_ACTIVE_VALIDATORS_MAINCHAIN} elements.")` from the verify hook?

Reply:
done